creating_packages/define_abi_compatibility.rst

```diff
@@ -701,6 +701,49 @@ generated with the ``recipe_revision_mode`` can be resolved if no package for th
             self.compatible_packages.append(p)
 
 
+.. _full_transitive_package_id:
+
+Enabling full transitivity in package_id modes
+++++++++++++++++++++++++++++++++++++++++++++++
+
+When a package declares in its ``package_id()`` method that it is not affected by its dependencies, that will propagate down
+to the indirect consumers of that header only package. If the header only package dependency is a static library, for example, then downstream
+consumers of the header only library that uses a package mode different from the default, should be also affected by the upstream
+transitivity dependency. Lets say that we have the following scenario:
+
+- ``pkgA/1.0``
+- ``pkgB/1.0`` depends on ``pkgA/1.0``, and defines ``self.info.header_only()`` in its ``package_id()``
+- ``pkgC/1.0`` depends only on ``pkgB/1.0``
+- ``App/1.0`` depends on ``pkgC/1.0`` and ``pkgA/1.0``
+- We are using ``full_version_mode``
+- Now we create a new ``pkgA/2.0`` that has some changes in its header, that would require to rebuild ``pkgC/1.0`` against it.
+- ``App/1.0`` now depends on ``pkgC/1.0`` and ``pkgA/2.0``
```

Comment:
We need a graph here, and we need to see the `conanfile.py` sources. This way it is really hard to understand for someone reading about package IDs

Reply:
Done